### PR TITLE
fix build

### PR DESCRIPTION
**Pull request by DmytroOrlov**

### CODE REVIEW

File: server/app/service/ScalaTestRunner.scala [hunk 1]
```diff
@@ -84,12 +84,10 @@ object ScalaTestRunner {
...
     val runningCode = s"$defaultImports; $suite; $patchedSolution; new $suiteName(new $userClass)"
 
-    tryExec {
-      execSuite(suiteInstance = tb.eval(tb.parse(runningCode)).asInstanceOf[Suite])
-    }
+    execSuite(suiteInstance = tb.eval(tb.parse(runningCode)).asInstanceOf[Suite])
   }
 
-  private def tryExec(suite: => String) =
+  def tryExec(suite: => String) =
```
Comment (novakov-alexey) [inline server/app/service/ScalaTestRunner.scala]: Do we use this method externally? Why did you remove private modificator?

File: server/app/service/SimpleWebSocketActor.scala [hunk 2]
```diff
@@ -75,14 +74,9 @@ object SimpleWebSocketActor {
...
     Future {
-      Try(createSolutionAndExec(solution, suiteClass, solutionTrait)) match {
-        case Success(s) =>
-          channel.pushNext(Line(s))
-          channel.pushComplete()
-        case Failure(e) =>
-          channel.pushNext(Line(s"Test $failedInRuntimeMarker with error:\n${e.getMessage}'"))
-          channel.pushComplete()
-      }
+      val lines = tryExecSuite(solution, suiteClass, solutionTrait).split("\n")
```
Comment (novakov-alexey) [inline server/app/service/SimpleWebSocketActor.scala]: Aha, I see it

File: server/test/service/ScalaTestDynamicRunnerTest.scala [hunk 3]
```diff
@@ -44,12 +45,11 @@ class ScalaTestDynamicRunnerTest extends ScalaTestRunnerTest with FlatSpecLike {
...
-  override def getReport(solution: String) = {
-    ScalaTestRunner.execSuite(
-      solution,
-      correctSuite
-    )
-  }
+  override def getReport(solution: String) =
+    tryExec {
+      ScalaTestRunner.execSuite(solution, correctSuite
+      )
```
Comment (novakov-alexey) [inline server/test/service/ScalaTestDynamicRunnerTest.scala]: move skobka up

File: server/app/service/ScalaTestRunner.scala (hunk 1 above)
Comment (DmytroOrlov) [inline server/app/service/ScalaTestRunner.scala]: I'm not using anymore, we can return private scope

File: server/app/service/ScalaTestRunner.scala [hunk 4]
```diff
@@ -26,7 +26,7 @@ object ScalaTestRunner {
   /**
    * Runs suite loaded in runtime with dynamic solution
    */
-  def execSuite(solution: String, suiteClass: Class[Suite], solutionTrait: Class[AnyRef]): String =
+  def tryExecSuite(solution: String, suiteClass: Class[Suite], solutionTrait: Class[AnyRef]): String =
```
Comment (novakov-alexey) [inline server/app/service/ScalaTestRunner.scala]: let's wrap with tryExec other methods below and rename this one back to execSuite to be consistent with other, because every API method may catch exception when we run reflection instance creation, so that each method should have tryExec inside. As already discussed these two methods are overloaded and they do the same thing. So that they better equal names

Comment (DmytroOrlov) [inline server/app/service/ScalaTestRunner.scala]: I don't need tryExec everywhere

File: server/app/service/ScalaTestRunner.scala [hunk 5]
```diff
@@ -89,9 +89,10 @@ object ScalaTestRunner {
     }
   }
 
-  private def tryExec(suite: => String) =
-    try suite catch {
-      case NonFatal(e) => s"Test $failedInRuntimeMarker with error:\n${e.getMessage}'"
+  private[service] def tryExec(suite: => String) =
+    Try(suite) match {
+      case Failure(e) => Failure(new SolutionException(s"Test $failedInRuntimeMarker with error:\n${e.getMessage}'", e))
+      case Success(v) => Success(v)
```
Comment (novakov-alexey) [inline server/app/service/ScalaTestRunner.scala]: It is smth I do not really like, but can't find our better solution for now
